src/components/browser/index.js

```diff
@@ -0,0 +1,217 @@
+        let versionRegExpResult = versionRegExp.exec(userAgent)
+
+        if (versionRegExpResult != null && versionRegExpResult[1])
+          setOsVersion(versionRegExpResult[1], (OS_DATA[i].versionSeparator) ? OS_DATA[i].versionSeparator : '.', osObject)
+
+      }
+      break
+    }
+  }
+  return osObject
+}
+
+// Set OS version
+const setOsVersion = function(version, separator, osObject) {
+  let finalSeparator = separator.substr(0, 1) == '[' ? new RegExp(separator, 'g') : separator
+  const splitVersion = version.split(finalSeparator, 2)
+
+  if (separator != '.') version = version.replace(new RegExp(separator, 'g'), '.')
+
+  osObject.fullVersion = version
+
+  // Major version
+  if (splitVersion && splitVersion[0])
+    osObject.majorVersion = parseInt(splitVersion[0])
+
+  // Minor version
+  if (splitVersion && splitVersion[1])
+    osObject.minorVersion = parseInt(splitVersion[1])
+}
+
+// Set viewport size
+export const getViewportSize = function() {
+  let viewportObject = {}
+
+  viewportObject.width = $(window).width()
+  viewportObject.height = $(window).height()
+
+  return viewportObject
+}
+
+// Set viewport orientation
+const setViewportOrientation = function() {
+  switch (window.orientation) {
+  case -90:
+  case 90:
+    Browser.viewport.orientation = 'landscape'
+    break
+  default:
+    Browser.viewport.orientation = 'portrait'
+    break
+  }
+}
+
+export const getDevice = function() {
+  let platformRegExp = Browser.isMobile ? new RegExp(';[^;]+;([^\)]+)') : new RegExp(';([^\)]+)') // eslint-disable-line no-useless-escape
+  let device = platformRegExp.exec(Browser.userAgent)[1].trim()
+  return device
+}
+
+const browserInfo = getBrowserInfo(navigator.userAgent)
+
+Browser.isEdge = /edge/i.test(navigator.userAgent)
+Browser.isChrome = /chrome|CriOS/i.test(navigator.userAgent) && !Browser.isEdge
+Browser.isSafari = /safari/i.test(navigator.userAgent) && !Browser.isChrome && !Browser.isEdge
+Browser.isFirefox = /firefox/i.test(navigator.userAgent)
+Browser.isLegacyIE = !!(window.ActiveXObject)
+Browser.isIE = Browser.isLegacyIE || /trident.*rv:1\d/i.test(navigator.userAgent)
+Browser.isIE11 = /trident.*rv:11/i.test(navigator.userAgent)
+Browser.isChromecast = Browser.isChrome && /CrKey/i.test(navigator.userAgent)
+Browser.isMobile = /Android|webOS|iPhone|iPad|iPod|BlackBerry|Windows Phone|IEMobile|Mobile Safari|Opera Mini/i.test(navigator.userAgent)
+Browser.isiOS = /iPad|iPhone|iPod/i.test(navigator.userAgent)
+Browser.isAndroid = /Android/i.test(navigator.userAgent)
+Browser.isWindowsPhone = /Windows Phone/i.test(navigator.userAgent)
+Browser.isWin8App = /MSAppHost/i.test(navigator.userAgent)
+Browser.isWiiU = /WiiU/i.test(navigator.userAgent)
+Browser.isPS4 = /PlayStation 4/i.test(navigator.userAgent)
+Browser.hasLocalstorage = hasLocalstorage()
+Browser.hasFlash = hasFlash()
+
+Browser.name = browserInfo.name
+Browser.version = browserInfo.version
```

Comment:
Now, this properties will be redundant with `Browser.data.name` and `Browser.data.fullVersion`. Maybe we should remove them?

Reply:
For reasons of backward compatibility, it would be better to leave both methods at this time and warn they will be removed in a next version (0.3.0)

Reply:
👍